src/misc_tools.h

```diff
@@ -190,4 +197,18 @@ bool is_ip4_address(const char *address);
  */
 bool is_ip6_address(const char *address);
 
+
+/*
+ * Frees `length` members of pointer array `arr` and frees `arr`.
+ */
+void free_ptr_array(void **arr, size_t length);
+
+/*
+ * Returns a new array of `length` pointers of size `ptr_size`. Each pointer is allocated `bytes` bytes.
```

Comment:
A pointer will always be the size of a pointer? No need to pass sizeof(void*) into the function.

Reply:
You could also at the cost of an extra 4/8 byte "null-terminate" the list so you wouldn't have to pass the length around everywhere.

Reply:
>A pointer will always be the size of a pointer? No need to pass sizeof(void*) into the function.

`length` is the length of the array/list. You're freeing the double pointer plus all the pointers that it points to.

>You could also at the cost of an extra 4/8 byte "null-terminate" the list so you wouldn't have to pass the length around everywhere.

That's a good idea.